apps/bajour/components/should-be-website-builder/sitemap.ts

```diff
@@ -0,0 +1,44 @@
+import {generateSitemap} from '@wepublish/website'
+import {ApiV1} from '@wepublish/website'
+import {NextApiRequest} from 'next'
+import getConfig from 'next/config'
+
+export const getSitemap = async (req: NextApiRequest): Promise<string> => {
+  const siteUrl = `https://${req.headers['host']}`
+
+  const generate = generateSitemap({
+    siteUrl,
+    title: 'Bajour'
+  })
+
+  const {publicRuntimeConfig} = getConfig()
+  const client = ApiV1.getV1ApiClient(publicRuntimeConfig.env.API_URL!, [])
+
+  const [{data: articleData}, {data: pageData}] = await Promise.all([
+    client.query({
+      query: ApiV1.ArticleListDocument,
+      variables: {
+        take: 50,
+        sort: ApiV1.ArticleSort.PublishedAt,
+        order: ApiV1.SortOrder.Descending
+      } as ApiV1.ArticleListQueryVariables
+    }),
+    client.query({
+      query: ApiV1.SitemapPageListDocument,
+      variables: {
+        take: 100,
+        sort: ApiV1.PublishedPageSort.PublishedAt,
+        order: ApiV1.SortOrder.Descending
+      } as ApiV1.PageListQueryVariables
+    })
+  ])
+
+  return generate(articleData.articles.nodes ?? [], [
+    `${siteUrl}/author`,
```

Comment:
I wonder if we could extract those from the router on the API side but couldn't find anything.